### PR TITLE
Phase 2 CSS improvements for consent banner and overlay

### LINKED ISSUE #4274: CSS customization - Phase 2

### PR DESCRIPTION
**Pull request by galvana**
Closes #4274 

### Description Of Changes

This PR focused on:
- Making the banners and overlays responsive
- Breaking up the modal into a header, body, and footer to be able to style custom headers and footers

I tried to make the layout changes with just CSS but there were enough differences between the desktop and mobile layouts that I think it made sense to have an `isMobile` prop to conditionally render the components instead of trying to nest them or modify them with CSS.

### Code Changes

* [ ] Added an `isMobile` state to the `ConsentBanner.tsx`
  * [ ] To position the button group in a different location based on the screen width
  * [ ] Change the manage preferences link to a button in the mobile view
* [ ] Refactored the `ConsentModal.tsx` to use `fides-modal-header`, `fides-modal-body`, and `fides-modal-footer`
* [ ] Misc CSS fixes

### Steps to Confirm

* [x] Configure fidesplus to use TCF
* [x] Add new systems with privacy declarations to enable the TCF experience
* [x] Start the Privacy Center
* [x] Navigate to http://localhost:3001/fides-js-demo.html to test the consent workflow
* [x] Navigate to http://localhost:3001/fides-js-demo.html?geolocation=fi-18 to test the TCF workflow
* [x] For both of these scenarios, resize the screen to see the mobile view

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

**Comment by galvana** [comment]
Hey @allisonking, heads up, I just made a tiny fix to the drop-down carets, they were being pushed out of the div if the corresponding text was too wide https://github.com/ethyca/fides/pull/4334/commits/279a9d16c3e7eb8d27952250410e42d531292447

**Comment by allisonking** [comment]
Got it, starting review now!

**Comment by allisonking** [comment]
small thing—could we add a lil bit of margin right to the iab tcf tag in small viewports? it seems to run into the legint toggle a bit
image

File: clients/fides-js/src/components/fides.css [hunk 1]
```diff
@@ -559,18 +577,28 @@ div#fides-banner-inner .fides-privacy-policy {
   width: 100%;
   display: flex;
   justify-content: space-between;
-  margin-right: 0.5em;
   min-height: 40px;
   align-items: center;
+
+  & svg {
```
Comment (allisonking) [inline clients/fides-js/src/components/fides.css:583]: I _think_ this kind of nesting doesn't work in firefox 😢 

see firefox on left, chrome on right
image

Comment (allisonking) [reply clients/fides-js/src/components/fides.css:583]: I'm on firefox 116 (which is a little out of date) which doesn't support it, seems to only be a recent thing https://caniuse.com/css-nesting

File: clients/fides-js/src/components/notices/NoticeOverlay.tsx [hunk 2]
```diff
@@ -128,12 +128,13 @@ const NoticeOverlay: FunctionComponent<OverlayProps> = ({
...
                 }
+                isMobile={isMobile}
               />
-            }
+            )}
           />
         ) : null
       }
-      renderModalContent={({ onClose }) => (
+      renderModalContent={() => (
```
Comment (allisonking) [inline clients/fides-js/src/components/notices/NoticeOverlay.tsx:137]: if this no longer needs an argument, I think it'd be cleaner to revert it back to a component, i.e.

```js
modalContent={
  <div>
  ...
  </div>
}
```

or even to let this be `children`

File: clients/fides-js/src/components/ConsentBanner.tsx [hunk 3]
```diff
@@ -24,9 +29,25 @@ const ConsentBanner: FunctionComponent<BannerProps> = ({
   bannerIsOpen,
   children,
   onVendorPageClick,
-  buttonGroup,
+  renderButtonGroup,
 }) => {
+  const [isMobile, setIsMobile] = useState(window.innerWidth < 768);
```
[the remark this answers is not in the document]
Comment (allisonking) [reply clients/fides-js/src/components/ConsentBanner.tsx:34]: this whole state + useEffect could go into `hooks.ts` to keep this component cleaner

**Review by allisonking — APPROVED** [review]
tested it out and it's looking nice, great work! a few comments but nothing blocking. largest thing might be the carets in firefox

**Comment by galvana** [comment]
> small thing—could we add a lil bit of margin right to the iab tcf tag in small viewports? it seems to run into the legint toggle a bit image

Nice catch! The tag should be left-aligned even for the small viewports
image

**Comment by galvana** [comment]
Thanks for the review @allisonking 🙏 I addressed the visual issues but in the interest of time, I'm moving the code-cleanup recommendations to the next sprint. I'll address them when I come back from break.

**Pull request merged**